tests/test_messages.py

```
@@ -521,16 +524,67 @@ def test_temp_file_cleaned_up_on_error(self, mock_applescript):
         self.assertEqual(len(leaked), 0, f"Temp files leaked: {leaked}")
 
+    @patch("mac_messages_mcp.messages.query_messages_db")
+    @patch("mac_messages_mcp.messages.run_applescript")
+    def test_temp_file_is_owner_only(self, mock_applescript, mock_query_db):
+        """mkstemp must create the message file as 0o600 before AppleScript reads it."""
+        import re
+        import stat
+
+        seen_mode = {}
+
+        def inspect_script(script):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `inspect_script`

Add return type annotation: `str`

(ANN202)